client/src/app/project/[projectid]/searchComponents/SearchResults.tsx

```diff
@@ -24,6 +24,57 @@ const SearchResults: FC<Props> = ({
   handleDrag,
   scheduleColors,
 }) => {
+  const [expandSort, setExpandSort] = useState(false);
+  const [expandFilter, setExpandFilter] = useState(false);
+  const [data, setData] = useState(noteData);
+  const [sorted, setSorted] = useState("");
+  const [filterValues, setFilterValues] = useState<string []>([]);
+  const [resort, setResort] = useState(false);
+
+  const sortByValue = (value: string) => {
+    if (value === "name") {
+      data.sort((a,b) => a.noteName.localeCompare(b.noteName));
+    }
+    else if (value === "date") {
+      data.sort((a,b) => a.scheduleDate ? b.scheduleDate ? a.scheduleDate - b.scheduleDate : -1 : 1);
+    }
+    else if (value === "priority") {
+      const order = { 'Low': 1, 'Medium': 2, 'High': 3 }
+      //Not sure why this error occurs
+      data.sort((a,b) => order[b.priority] - order[a.priority]);
+    }
```

Comment:
This is a great catch! The data is typed as `NoteData[]` in `util/models/ProjectModels.ts`
```
export interface NoteData {
  noteName: string,
  locationID: string,
  formattedAddress: string,
  customNote?: string,
  openHours?: string,
  closeHours?: string,
  picture?: string,
  scheduleDate?: number,
  priority: string,
}
```

Under priority is it typed as a `string` which can be anything tbh. If we defined the interface properly it will fix the typescript error:
```
type priorityValues = "Low" | "Medium" | "High"

export interface NoteData {
  noteName: string,
  locationID: string,
  formattedAddress: string,
  customNote?: string,
  openHours?: string,
  closeHours?: string,
  picture?: string,
  scheduleDate?: number,
  priority: priorityValues,
}
```

Reply:
This fixed it beautifully thanks